mantis-runtime-executor/src/main/java/io/mantisrx/server/worker/RuntimeTaskImpl.java

```diff
@@ -71,9 +72,15 @@ public class RuntimeTaskImpl extends AbstractIdleService implements RuntimeTask
     private Optional<Job> mantisJob = Optional.empty();
 
     private ExecuteStageRequest executeStageRequest;
+    private FailoverStatusClient failoverStatusClient;
 
     public RuntimeTaskImpl() {
+        this(FailoverStatusClient.DEFAULT);
+    }
+
+    public RuntimeTaskImpl(FailoverStatusClient failoverStatusClient) {
         this.tasksStatusSubject = PublishSubject.create();
+        this.failoverStatusClient = failoverStatusClient;
     }
```

Comment:
I think this will work fine on internal nfmantis wrapper but on OSS the instance is built from SingleTaskOnlyFactory so there is no path yet to inject this client to OSS yet?

Reply:
I think it's ok for now but probably better to document this somewhere explicitly.